Source/SwiftLintBuiltInRules/Rules/Performance/EmptyCountRule.swift

```diff
@@ -59,6 +97,67 @@ private extension EmptyCountRule {
             }
         }
     }
+
+    final class Rewriter: ViolationsSyntaxRewriter {
+        private let operators: Set = ["==", "!=", ">", ">=", "<", "<="]
+        private let configuration: EmptyCountConfiguration
+
+        init(configuration: EmptyCountConfiguration,
+             locationConverter: SourceLocationConverter,
+             disabledRegions: [SourceRange]) {
+            self.configuration = configuration
+            super.init(locationConverter: locationConverter, disabledRegions: disabledRegions)
+        }
+
+        override func visit(_ node: SequenceExprSyntax) -> ExprSyntax {
+            guard let folded = try? OperatorTable.standardOperators.foldSingle(node) else { return super.visit(node) }
+
+            if let infix = folded.as(InfixOperatorExprSyntax.self) {
+                return visit(infix)
+            } else {
+                return super.visit(folded)
+            }
+        }
+
+        override func visit(_ node: InfixOperatorExprSyntax) -> ExprSyntax {
+            guard let operatorNode = node.operator.as(BinaryOperatorExprSyntax.self),
+                  let binaryOperator = operatorNode.operator.binaryOperator,
+                  operators.contains(binaryOperator) else {
+                return super.visit(node)
+            }
+
+            if let (count, position) = node.countNodeAndPosition(onlyAfterDot: configuration.onlyAfterDot) {
+                let newNode: ExprSyntax? = if let count = count.as(MemberAccessExprSyntax.self) {
+                    count.with(\.declName.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+                } else if let count = count.as(DeclReferenceExprSyntax.self) {
+                    count.with(\.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+                } else {
+                    nil
+                }
+
+                if let newNode {
+                    correctionPositions.append(position)
+                    if ["!=", "<", ">"].contains(binaryOperator) {
+                        return newNode.negatedExprSyntax
+                            .withTrivia(from: node)
+                    } else {
+                        return ExprSyntax(newNode)
+                            .withTrivia(from: node)
+                    }
+                } else {
+                    let left = node.leftOperand.is(InfixOperatorExprSyntax.self)
```

Comment:
There is no test for this branch. Please add some examples.

Reply:
Fixed at 7695b56

Since there was a test for a case where left and right are both infix, adding two others where only one of them is infix should be sufficient.

Reply:
I checked this be deleting the code. All tests are still successful. There might be more examples needed, or this case is just not possible to reach.

Reply:
This was indeed not possible to reach since `countNodeAndPosition()` and `node.hasBinaryOperator` would guard such cases before the `if` statement making it always succeed. The case this `else` statement was supposed to handle should be handled by future calls to `visit(_ node: InfixOperatorExprSyntax)` and I added a new test case with nested `InfixOperatorExprSyntax`es at 4a9314e